### ISSUE TITLE
LTI1.3 Provider Tool (OEX as Provider)

### ISSUE THREAD

**Issue by jmakowski1123** [openedx/platform-roadmap#203]
There's a need for the creation of a new, stable LTI1.3 Provider tool, where OEX serves as the LTI Provider. While this need is captured at a high level in the [Modular Learning user stories ](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3488677889/User+Stories+-+WIP)currently underway, it's likely a spin-off project in and of itself, with first steps being:

- [ ] Gathering perspectives from stakeholders across the community - what's the scope of the tool?
- [ ] Approach spec with community feedback loops

**Comment by felipemontoya** [comment]
@jmakowski1123 we are very interested in writing LTI1.3 support outside of the context of content libraries. This would be the card that represents that work, correct?

**Comment by antoviaque** [comment]
FYI @giovannicimolin - if you have some time to help review / advise on this, that would be a nice contribution :)

**Comment by giovannicimolin** [comment]
@antoviaque Yup - will do! Let me catch up on the discussion here. I'll reply to this tomorrow.

**Comment by giovannicimolin** [comment]
@jmakowski1123 This is great!

>Gathering perspectives from stakeholders across the community - what's the scope of the tool?

I think that the main goal of an "LTI Provider" (or if we use Ed1Tech terms: "LTI Tool") is to allow external tools to link content from our LMS on 3rd party LTI consumers (such as other LMS's). The goal of the project should be to implement a full set of LTI capabilities to Open edX: LTI 1.3 launches that can embed our LMS and the three main LTI Advantage services (Deep Linking, Grades, and the NRPS).

From the technical standpoint, this new app _should probably be_ a **pluggable app, installed by default but disabled in an Open edX instance.** Not every instance will need LTI 1.3 tool capabilities, but I think it's something important enough to be easy to turn on/off.

----

@felipemontoya @jmakowski1123 The document linked by @kuipumu contains a lot of great references and resources to get this work rolling.

A while ago there was a contribution (https://github.com/openedx/edx-platform/pull/21435) to implement LTI 1.3 tool capability in the platform with a working prototype. So, I think that the quickest way to get this done would be to re-look at `pylti1.3` and implement that PR as a pluggable app - checking if the approach used them will also fit with the new OEPs and ADRs.

[The fork of that PR](https://github.com/CredoEducation/edx-platform ) seems to have evolved LTI 1.3 tool capabilities a bit more since then. Also, https://github.com/dmitry-viskov/pylti1.3 seems to be well-maintained and is already being used on content libraries, so this is overall a good starting point for this work.

**Comment by scottrish** [comment]
There is a claim (https://purl.imsglobal.org/spec/lti-bo/claim/basicoutcome) to allow [Basic Outcomes 1.1](https://www.imsglobal.org/spec/lti-bo/v1p1) params to be included in 1.3 messages, however as Basic Outcomes is deprecated do we have any insight into whether 1.3 platform implementations generally support Basic Outcomes or have they largely migrated to [Assignment & Grade Services](https://www.imsglobal.org/spec/lti-ags/v2p0/).
For reference, the current Open edX 1.3 platform implementation does not appear to support Basic Outcomes.  
I wouldn't want to invest time building in support for Basic Outcomes if most 1.3 platforms won't utilize it. Maybe for the most commonly encountered cases (Canvas, Blackboard, Moodle, D2L) we could try to check whether their 1.3 platform supports basic outcomes? @jmakowski1123 ?

**Comment by scottrish** [comment]
@jmakowski1123 Updated ADR - https://github.com/Pearson-Advance/openedx-lti-tool-plugin/blob/vue/PADV-303/docs/decisions/0001-launch-complete-course.rst

The team are looking for feedback on the proposed and rejected approaches to launching an entire course via LTI.  We are moving forward with approach 4 but are open to adjusting that based on feedback.

**Comment by jmakowski1123** [comment]
Thanks @scottrish ! Flagging @ormsbee @kdmccormick @e0d  for feedback on the ADR
